src/core/pagerankGraph.test.js

```diff
@@ -175,4 +175,58 @@ describe("core/pagerankGraph", () => {
       checkProbabilityDistribution(pg);
     });
   });
+
+  describe("equals", () => {
+    it("PagerankGraph is equal to itself", () => {
+      const pg = examplePagerankGraph();
+      expect(pg.equals(pg)).toBe(true);
+    });
+    it("two identicalPagerankGraphs are equal", () => {
+      const pg1 = examplePagerankGraph();
+      const pg2 = examplePagerankGraph();
+      expect(pg1.equals(pg2)).toBe(true);
+    });
+    it("unequal syntheticLoopWeight => unequal", () => {
+      const pg1 = new PagerankGraph(nonEmptyGraph(), defaultEvaluator, 0.1);
+      const pg2 = new PagerankGraph(nonEmptyGraph(), defaultEvaluator, 0.2);
+      expect(pg1.equals(pg2)).toBe(false);
+    });
+    it("unequal graph => unequal", () => {
+      const pg1 = new PagerankGraph(nonEmptyGraph(), defaultEvaluator, 0.1);
+      const g2 = nonEmptyGraph().addNode(NodeAddress.empty);
+      const pg2 = new PagerankGraph(g2, defaultEvaluator, 0.2);
```

Comment:
Are you wanting to isolate the difference in graph nodes here or is this meant to be a catch-all for different kinds of inequalities? Perhaps keep the synthetic loop weights equal if its the former.

Reply:
Good catch! I intended to just look at the difference in the underlying graph, putting different SLW was a mistake. Will fix.